### PR TITLE
Download java modal

### PR DESCRIPTION
**Pull request by benrudolph**
@orangejenny this fixes up a few issues with the deploy modal. best read commit-by-commit. in essence, since we were using ids in the modal and multiple modals were being generated, the modal would behave weirdly by not opening after a second modal was opened or not properly closing other modals in the group. this should fix all those issues

http://manage.dimagi.com/default.asp?232280#1184182

interrupt: @calellowitz @sravfeyn

### CODE REVIEW

File: corehq/apps/app_manager/static/app_manager/js/releases.js [hunk 1]
```diff
@@ -37,6 +37,11 @@ hqDefine('app_manager/js/releases.js', function () {
             self.generating_url(false);
         };
 
+        self.onSMSPanelClick = function() {
+            track_deploy_type('Send to phone via SMS');
```
Comment (dimagimon) [inline corehq/apps/app_manager/static/app_manager/js/releases.js]: 'track_deploy_type' is not defined. (no-undef)

File: corehq/apps/app_manager/static/app_manager/js/releases.js [hunk 2]
```diff
@@ -37,6 +37,11 @@ hqDefine('app_manager/js/releases.js', function () {
...
+            self.generate_short_url('short_url');
+        }
```
Comment (dimagimon) [inline corehq/apps/app_manager/static/app_manager/js/releases.js]: Missing semicolon. (semi)

File: corehq/apps/app_manager/static/app_manager/js/releases.js [hunk 3]
```diff
@@ -31,12 +32,21 @@ hqDefine('app_manager/js/releases.js', function () {
             return profiles;
         };
 
+        self.track_deploy_type = function(type) {
+            ga_track_event('App Manager', 'Deploy Type', type);
+        }
```
Comment (dimagimon) [inline corehq/apps/app_manager/static/app_manager/js/releases.js]: Missing semicolon. (semi)

**Comment by orangejenny** [comment]
Looks good. Not sure why tests haven't run, but I don't think there's test coverage for this anyway.

**Comment by benrudolph** [comment]
they ran on the previous commit. and they were successful!

**Comment by orangejenny** [comment]
Indeed, I misread the UI. Excellent.

**Pull request merged**